libgstc/c/libgstc.c

```diff
@@ -423,6 +423,42 @@ gstc_pipeline_get_graph (GstClient * client, const char *pipeline_name,
   return ret;
 }
 
+GstcStatus
+gst_pipeline_get_state (GstClient * client, const char *pipeline_name,
+    char **out)
+{
+  GstcStatus ret = GSTC_OK;
+  int asprintf_ret;
+  char *what;
+  char *response;
+
+  gstc_assert_and_ret_val (client != NULL, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (pipeline_name != NULL, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (out != NULL, GSTC_NULL_ARGUMENT);
+
+  asprintf_ret = asprintf (&what, PIPELINE_STATE_FORMAT, pipeline_name);
+  if (PRINTF_ERROR == asprintf_ret) {
+    return GSTC_OOM;
+  }
+
+  ret = gstc_cmd_read (client, what, &response, client->timeout);
+  if (ret != GSTC_OK) {
+    goto unref;
+  }
+
+  ret = gstc_json_child_string (response, "response", "value", out);
+  if (ret != GSTC_OK) {
+    goto unref_response;
```

Comment:
This goto and the label are not really necessary. The program flow goes to  `unref_response` anyways, even when there is no error.